cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/ArtifactInspector.java

```diff
@@ -427,8 +430,12 @@ private MetadataMutation getMetadataMutation(PluginId pluginId, Class<?> cls) {
     if (tags.isEmpty() && properties.isEmpty()) {
       return null;
     }
-    return new MetadataMutation.Create(pluginId.toMetadataEntity(),
-                                       new io.cdap.cdap.spi.metadata.Metadata(metadataScope, tags, properties),
+    MetadataEntity metadataEntity = pluginId.toMetadataEntity();
+    // validate the tags and properties
+    metadataValidator.validateTags(metadataEntity, tags);
+    metadataValidator.validateProperties(metadataEntity, properties);
```

Comment:
does this bubble all the way up and result in a 400 error code when deploying a plugin artifact? 

Does it result in an actionable error message? It seems like we would need to enhance the message to include the problematic plugin, otherwise it will be difficult to know what's invalid.

Reply:
Yes, this will result in a 400. 

The `InvalidMetadataException` already includes entity that is invalid with good error message, I added the getDescription to handle the plugin well, seems like it was missing last time we introduced plugin metadata